src/main/java/appeng/tile/storage/TileChest.java

```diff
@@ -572,87 +579,46 @@ public void onChangeInventory( final IInventory inv, final int slot, final InvOp
 				this.markForUpdate();
 			}
 		}
-	}
-
-	@Override
-	public boolean canInsertItem( final int slotIndex, final ItemStack insertingItem, final EnumFacing side )
-	{
-		if( slotIndex == 1 )
+		if( inv == inputInventory )
 		{
-			if( AEApi.instance().registries().cell().getCellInventory( insertingItem, this, StorageChannel.ITEMS ) != null )
-			{
-				return true;
-			}
-			if( AEApi.instance().registries().cell().getCellInventory( insertingItem, this, StorageChannel.FLUIDS ) != null )
+			if( mc == InvOperation.insert )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1066)